3/test/excercise3.spec.js

```diff
@@ -0,0 +1,110 @@
+            expect(result.data.assignees.length).toBe(3);
+        })
+    })
+})
+
+describe('Query: Given todos with different states', () => {
+
+    const queryTodosByState = gql`query getTodosByState($state: State!) { todosByState(state: $state) {id, message} }`;
+
+    it('return done todos', () => {
+        query({query: queryTodosByState, variables: {state: 'DONE'}}).then((result) => {
+            expect(result.data.todosByState.length).toBe(3);
+        })
+    })
+    it('return undone todos', () => {
+        query({query: queryTodosByState, variables: {state: 'UNDONE'}}).then((result) => {
+            expect(result.data.todosByState.length).toBe(2);
+        })
+    })
+    it('return in progress todos', () => {
+        query({query: queryTodosByState, variables: {state: 'IN_PROGRESS'}}).then((result) => {
+            expect(result.data.todosByState.length).toBe(1);
+            expect(result.data.todosByState[0].message).toBe('Go shopping')
+        })
+    })
+})
+
+describe('Mutation: Given a new todo', () => {
+
+    const mutateCreateTodo = gql`mutation createNewTodo($id: ID, $message: String!, $state: State, $assigneeId: ID) {
+        createTodo(id: $id, message: $message, state: $state, assigneeId: $assigneeId) { id, message } 
+    }`;
+
+    describe('Add todo item with existing unique id to list', () => {
+        it('reject duplicate id entry', () => {
+            mutate({mutation: mutateCreateTodo, variables: 
+                {id: 1, message: 'Do something', state: "UNDONE", assigneeId: 1}}).then((result) => {
+                expect(result.data.createTodo).toBeNull();
+            })
+        })
+    })
+    describe('Add todo item with new unique id to list', () => {
+        it('todo item was added to list', () => {
+            mutate({mutation: mutateCreateTodo, variables: 
+                {id: 23, message: 'Do something', state: 'UNDONE', assigneeId: 1}}).then((result) => {
+                expect(result.data.createTodo).not.toBeNull();
+                expect(result.data.createTodo.id).toBe('23');
+                expect(result.data.createTodo.message).toEqual('Do something');
+            })
+        })
+    })
+})
+
+describe('Mutation: Update an existing todo', () => {
+    
+    const mutateUpdateTodo = gql`mutation updateTodo($id: ID!, $message: String, $state: State, $assigneeId: ID) {
+        updateTodo(id: $id, message: $message, state: $state, assigneeId: $ assigneeId) { id, message }}`;
+    const queryTodo = gql`query getTodo($id: ID!) { todo(id: $id) { id, message, state, assigneeId } }`;
+
+    it('todo item is updated', () => {
+        mutate({mutation: mutateUpdateTodo, variables: {id: 1, message: 'New TODO', state: 'IN_PROGRESS', assigneeId: 3}}).then((result) => {
+            expect(result.data.updateTodo).toBeTruthy();
+        })
+        query({query: queryTodo, variables: {id: 1}}).then((result) => {
```

Comment:
You can make sure that a change was introduced by the call of `mutate` by running:
```js
// expect that query() returns the initial item
// do something, e. g. mutate()
// expect that query() returns the updated item
// that way you can make sure the change is related to your action in the test
```